docs/parse_argparse.py

```diff
@@ -0,0 +1,315 @@
+    # extract the "kmers" of kmers.py
+    basename = os.path.splitext(fname)[0]
+    fname_rst = basename + ".rst"  # make it kmers.rst
+    # another temporary file to store stdout
+    __, path_tmp_rst = tempfile.mkstemp()
+    cmd = f"python {tmp_fpath} -h"
+    # this block writes the argparse output to the respective ".rst" file with proper indentation
+    # this is the output that will be under "..code-block::" and will be displayed in html
+    with open(path_tmp_rst, "w+") as stdout:
+        # capture the argparse output
+        subprocess.call(cmd, stdout=stdout, shell=True)
+        # takes the cursor to beginning to copy and indent stdout.
+        stdout.seek(0)
+        # The indentation is done because the arparse text needs to be indented
+        # under the ..code-block:: section in the final rst file
+        # https://docs.python.org/2/library/textwrap.html#textwrap.TextWrapper
+        wrapper = textwrap.TextWrapper(
+            initial_indent="\t", subsequent_indent="\t", width=80)
+        wrapped = ""
+        path_script_rst = os.path.join(
+            rst_dirpath, fname_rst)
+        # now the path is "docs/source/scripts/comon/kmers.rst"
+        with open(path_script_rst, "w") as fh_rst:
+            write_text = "\n".join(["="*len(fname), fname, "=" *
+                                    len(fname), " ", ".. code-block:: shell", "\n"])
+            fh_rst.write(write_text)
+            for line in stdout:
+                # copying the indented text
+                wrapped += wrapper.fill(line) + "\n"
+            # writing the indented text to the final rst files, this will be dislayed in html
+            fh_rst.write(wrapped)
+        os.remove(path_tmp_rst)
+
+
+def design_main_file(fpath, rst_dirpath):
+    """
+    Creates a _main file for each directory, and link sub-directories with the toc tree of parent directory.
+    Eg. Linking external with the toctree of common
+    These rst files will be called by the toctreee in scripts_main.rst
+
+    Parameters
+    ----------
+    fpath : string
+        file path that needs to be documented,
+        e.g. '</path/to/autometa/script.py>'
+    rst_dirpath : string
+        path to the directory where script.rst files will be written,
+        e.g. '</path/to/docs/source/scripts/directory>'
+    """
+    # extract "/home/siddharth/Autometa/autometa/common/external"
+    dir_path = os.path.dirname(os.path.abspath(fpath))
+    rst_main_dir = os.path.basename(dir_path)  # extraxt "external"
+    rst_main_file = rst_main_dir + "_main.rst"  # making a file external_main
+    path_main_file = os.path.join(rst_dirpath, rst_main_file)
+    outlines = ""
+    if not os.path.exists(path_main_file):
+        with open(path_main_file, "w") as fh:
+            lines = "\n".join(["="*len(rst_main_dir), rst_main_dir, "="*len(rst_main_dir), " ", ".. toctree::",
+                               "\t :maxdepth: 2", "\t :caption: Table of Contents", " "])
```

Comment:
This is extremely difficult to read. Again, you can use f-strings here to improve readabillity and you can similarly place the formatting text into a variable. You could also place this into its own function similar to before. Although, not entirely necessary

```python
def get_toctree_header(dirname):
    formatting = "=" * len(dirname)
    header = f'{formatting}\n{dirname}\n{formatting}\n.. toctree::'
    header +=  "\n\t :maxdepth: 2\n\t :caption: Table of Contents\n"
    return header

````

```suggestion
            formatting = "="*len(rst_main_dir)
            toctree_header = f'{formatting}\n{rst_main_dir}\n{formatting}\n.. toctree::'
            toctree_header +=  "\n\t :maxdepth: 2\n\t :caption: Table of Contents\n"
```